### PR TITLE
Adds a variant of fake handcuffs to the traitor uplink

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

This PR adds a a special variant of the fake handcuffs toy to the traitor uplink as a 1 TC item. Unlike the toy version, these professional-grade fake handcuffs have the same name and description that normal handcuffs do, and the only noticeable difference between these fake handcuffs and "real" handcuffs is how long it takes to break out of the fake handcuffs: 1 second.

## Why It's Good For The Game

You can do some pretty neat things with fake handcuffs, but getting them is RNG dependent and requires some grinding at an arcade machine or spending a lot of money on toy crates.

You can, for example, handcuff yourself before Sec gets to you, then break out of your cuffs after the officer who brought you into Sec drops his guard (you can do the same thing with a freedom implant, but those things are expensive, yo). You can use these as normal handcuffs in a pinch if you really need some, as long as you keep your victim moving. You can do a Houdini act with them. You can "arrest" a fellow traitor, then reenact that scene from Episode IV with them. You can make secbots set to arrest mode and turrets set to stun mode stop attacking you/not attack you. Clown ops can quickly uncuff themselves if they accidentally cuff themselves with them. You can probably do more clever things with these than you can with a deck of syndicate playing cards, which costs 2 TC instead of 1 TC.

## Changelog
:cl: ATHATH
add: Adds professional-grade fake handcuffs to the traitor uplink.
balance: You can now handcuff yourself instantly and silently with fake handcuffs.
fix: One-handed clowns can no longer accidentally cuff themselves (successfully).
fix: A handcuffing noise is now made when a clown accidentally handcuffs themselves.
tweak: The feedback given when you fail to handcuff someone has been improved.
fix: Starting an attempt to break out of trick handcuffs no longer makes you unable to click on anything for 10 seconds, even after you finish breaking out of your trick handcuffs.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Some alternate versions of this PR that I had considered:
- Making this a clown/mime-only traitor item.
- Not adding this new item to the game or to the traitor uplink and instead just letting people print fake handcuffs from autolathes.

Do either of these ideas sound better to you?

**Comment by Pumpkinoe** [comment]
I like the idea of it being an uplink item, but if you make it a clown/mime-only traitor item then it loses its surprise, because no sec officer with two brain cells is going to think "Oh wow, the clown's already cuffed, better bring him into sec!" without considering the clowncuffs.
I'd say keep it as an uplink item for all tator tots, because it sounds hilarious and fun.

**Comment by Fikou** [comment]
ah yes, fake fake handcuffs

**Comment by kriskog** [comment]
>You can probably do more clever things with these than you can with a deck of syndicate playing cards, which costs 2 TC instead of 1 TC.

Then why are these 1 TC 🤔

**Comment by Pumpkinoe** [comment]
probably because I don't think anyone would buy them if they were 2 TC
maybe someone should reprice the syndicate playing cards since you can literally find those for free in certain parts of certain maps

**Comment by PKPenguin321** [comment]
the cards to this is like the revolver to the emag
you can do cooler stuff with an emag but it costs less because you cant just straight drop people with it

**Comment by ATH1909** [comment]
Well, fuck, this is apparently a thing: https://github.com/tgstation/tgstation/issues/48318

Should I just close this PR, then?

**Comment by ATH1909** [comment]
Should I make applying these cuffs to yourself (but not to others) instant?

**Comment by ExcessiveUseOfCobblestone** [comment]
These types of items are what I like to see in the core uplink, much more than "It's a weapon but it DOES SOMETHING THAT NO OTHER WEAPON DOES: DAMAGES THEM WITH DIFFERENT FLAVOR!"

**Comment by MrDoomBringer** [comment]
>Should I make applying these cuffs to yourself (but not to others) instant?
Yes, absolutely. Would make quickly cuffing youself in the middle of a chase possible.
 
Also, make resisting out of them give you the cuffs in-hand, rather than drop them on the floor. That way it's easier to re-use the fake handcuffs you spent good gamer money on.

**Comment by ATH1909** [comment]
Alright, I've added the instant (and silent) self-handcuffing part, but it looks like making them appear in your hands after you break out of them might be a bit too risky (bug-wise) for me to code (plus, it makes breaking out of them look more "genuine").

Toy fake handcuffs are now also a subtype of traitor fake handcuffs, so I (and future coders) don't/won't have to write the same procs for both of them.

Finally, I noticed that clumsy people could cuff themselves despite only having one arm, but only if they accidentally cuffed themselves while trying to cuff someone. I've added some more checks to apply_cuffs() that should hopefully fix that issue.

**Comment by ATH1909** [comment]
I still need to fix #48318...

File: code/game/objects/items/handcuffs.dm [hunk 1]
```diff
@@ -146,9 +154,17 @@
 	icon_state = "handcuffAlien"
 
 /obj/item/restraints/handcuffs/fake
+	breakouttime = 1 SECONDS //takes only 1 second to break out of these
+
+/obj/item/restraints/handcuffs/fake/attack(mob/living/carbon/C, mob/living/user)
+	if(C == user)
+		apply_cuffs(user, user) //instant self-application
+	else
+		..()
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]:
```suggestion
		apply_cuffs(user, user) //instant self-application
		return
	return ..()
```

File: code/game/objects/items/handcuffs.dm [hunk 2]
```diff
@@ -74,12 +74,20 @@
 			else
 				to_chat(user, "<span class='warning'>You fail to handcuff [C]!</span>")
 		else
-			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>")
+			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>") //if you're trying to cuff yourself while one-handed, this will refer to you in the third person, which is a bit weird, but not a big enough issue to write some shitcode to keep that from happening
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]:
```suggestion
			to_chat(user, "<span class='warning'>[C == user ? "You don't" : "[C] doesn't"] have two hands...</span>")
```

File: code/game/objects/items/handcuffs.dm [hunk 3]
```diff
@@ -74,12 +74,20 @@
...
 
 /obj/item/restraints/handcuffs/proc/apply_cuffs(mob/living/carbon/target, mob/user, dispense = 0)
+
+	if(!istype(target)) //some sanity checks, in case apply_cuffs() gets called without attack() being called/fully run through (looking at YOU, clumsy people)
+		return
+
 	if(target.handcuffed)
 		return
 
+	if(!(target.get_num_arms(FALSE) >= 2 || target.get_arm_ignore()))
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]:
```suggestion
	if(!(target.get_num_arms(FALSE) >= 2) || target.get_arm_ignore())

```

File: code/game/objects/items/handcuffs.dm [hunk 4]
```diff
@@ -74,12 +74,20 @@
 			else
 				to_chat(user, "<span class='warning'>You fail to handcuff [C]!</span>")
 		else
-			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>")
+			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>") //if you're trying to cuff yourself while one-handed, this will refer to you in the third person, which is a bit weird, but not a big enough issue to write some shitcode to keep that from happening
 
 /obj/item/restraints/handcuffs/proc/apply_cuffs(mob/living/carbon/target, mob/user, dispense = 0)
+
+	if(!istype(target)) //some sanity checks, in case apply_cuffs() gets called without attack() being called/fully run through (looking at YOU, clumsy people)
+		return
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]: This shouldn't happen. Clumsy people are `carbon` as per the proc.
Avoid sanity checks that silently return. If something like this fails we want it to scream so we know where to fix.

Comment (ATH1909) [reply code/game/objects/items/handcuffs.dm]: A dextrous stand or a gorilla could click on themselves with fake handcuffs, no?

Comment (Rohesie) [reply code/game/objects/items/handcuffs.dm]: It should produce no result:
```
/obj/item/restraints/handcuffs/attack(mob/living/carbon/C, mob/living/user)
	if(!istype(C))
		return
```

**Comment by ATH1909** [comment]
Alright, I've made a lazy patch for the 10 second click cooldown applied whenever you try to break out of fake handcuffs. This isn't gonna affect anything else that invokes that 10 second click cooldown, just handcuffs (and unbuckling yourself from chairs and stuff while handcuffed).

If y'all are okay with it, I can change CLICK_CD_BREAKOUT to be 10 (deciseconds) instead of 100 (deciseconds). This would have the side effect of allowing people to start a new cuff breaking attempt once every second instead of once every 10 seconds, but I think that that's okay.

File: code/modules/uplink/uplink_items.dm [hunk 5]
```diff
@@ -1884,7 +1884,7 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/storage/fancy/cigarettes/cigpack_syndicate
 	cost = 2
 	illegal_tech = FALSE
-
+
```
Comment (Rohesie) [inline code/modules/uplink/uplink_items.dm]:
```suggestion
 
```

File: code/game/objects/items/handcuffs.dm [hunk 6]
```diff
@@ -74,10 +74,14 @@
 			else
 				to_chat(user, "<span class='warning'>You fail to handcuff [C]!</span>")
 		else
-			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>")
+			to_chat(user, "<span class='warning'>[C == user ? "You don't" : "[C] doesn't"] have two hands...</span>")
 
 /obj/item/restraints/handcuffs/proc/apply_cuffs(mob/living/carbon/target, mob/user, dispense = 0)
-	if(target.handcuffed)
+	if(target.handcuffed) //some sanity checks, in case apply_cuffs() gets called without attack() being called/fully run through (looking at YOU, clumsy people)
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]:
```suggestion
	if(target.handcuffed)
```
Comment no longer relevant, correct?

Comment (ATH1909) [reply code/game/objects/items/handcuffs.dm]: nah, target.handcuffed is still a sanity check, isn't it?

Comment (Rohesie) [reply code/game/objects/items/handcuffs.dm]: What do clumsy people have to do with it?

Comment (ATH1909) [reply code/game/objects/items/handcuffs.dm]: They bypass the target.handcuffed check in attack(), so it needs to be re-checked here.

**Comment by ATH1909** [comment]
Alright, I made some logic and feedback improvement changes so that it's more obvious to players why their cuffing attempts failed.

Also, accidentally cuffing yourself as a clown now makes a handcuffing sound (cuffing yourself with fake handcuffs still (intentionally) doesn't make a handcuffing sound, though).

File: code/game/objects/items/handcuffs.dm [hunk 7]
```diff
@@ -61,28 +61,39 @@
...
 			else
 				to_chat(user, "<span class='warning'>You fail to handcuff [C]!</span>")
 		else
-			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>")
+			to_chat(user, "<span class='warning'>[C == user ? "You don't" : "[C] doesn't"] have two hands...</span>")
+	else
+		to_chat(user, "<span class='warning'>[C == user ? "You're" : "[C] is"] already handcuffed!</span>")
 
-/obj/item/restraints/handcuffs/proc/apply_cuffs(mob/living/carbon/target, mob/user, dispense = 0)
+/obj/item/restraints/handcuffs/proc/apply_cuffs(mob/living/carbon/target, mob/user, dispense = 0, intentional = 0)
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]:
```suggestion
/obj/item/restraints/handcuffs/proc/apply_cuffs(mob/living/carbon/target, mob/user, dispense = FALSE, intentional = FALSE)

```

File: code/game/objects/items/handcuffs.dm [hunk 8]
```diff
@@ -61,28 +61,39 @@
...
 	if(target.handcuffed)
+		if(!intentional) //if a clumsy person rolls the chance to accidentally cuff themself but their character can't be cuffed, just give feedback as if they didn't make the (accidental) self-cuffing attempt at all and pretend that they just fiddled around with the cuffs without doing anything or something
+			to_chat(user, "<span class='warning'>[target == user ? "You're" : "[target] is"] already handcuffed!</span>")
+		return
+
+	if(!(target.get_num_arms(FALSE) >= 2 || target.get_arm_ignore()))
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]:
```suggestion
	if(target.get_num_arms(FALSE) < 2 && !target.get_arm_ignore())
```

File: code/game/objects/items/handcuffs.dm [hunk 9]
```diff
@@ -61,28 +61,40 @@
...
-				log_combat(user, C, "handcuffed")
 			else
 				to_chat(user, "<span class='warning'>You fail to handcuff [C]!</span>")
 		else
-			to_chat(user, "<span class='warning'>[C] doesn't have two hands...</span>")
+			to_chat(user, "<span class='warning'>[C == user ? "You don't" : "[C] doesn't"] have two hands...</span>")
+	else
+		to_chat(user, "<span class='warning'>[C == user ? "You're" : "[C] is"] already handcuffed!</span>")
+
+/obj/item/restraints/handcuffs/proc/apply_cuffs(mob/living/carbon/target, mob/user, dispense = FALSE, intentional = FALSE)
```
Comment (Rohesie) [inline code/game/objects/items/handcuffs.dm]: This will need documentation.

File: code/modules/uplink/uplink_items.dm [hunk 10]
```diff
@@ -1884,7 +1884,7 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
```
Comment (Rohesie) [inline code/modules/uplink/uplink_items.dm]:
```suggestion

```

File: code/game/objects/items/handcuffs.dm (hunk 9 above)
Comment (ATH1909) [reply code/game/objects/items/handcuffs.dm:76]: is the documentation that's here insufficient?

Comment (Rohesie) [reply code/game/objects/items/handcuffs.dm:76]: The proc has no documentation.
Here's the guide:
https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md

**Comment by MrDoomBringer** [comment]
>:b:erge conflict

**Comment by optimumtact** [comment]
Don't really see a need for this

**Comment by ATH1909** [comment]
The merge conflict resolver is being... "temperamental".

Probably because of the repo temporarily being down...

**Comment by ExcessiveUseOfCobblestone** [comment]
26 days to fix documentation

**Pull request closed without merging**